mesheryctl/internal/cli/root/registry/generate.go

```diff
@@ -181,11 +183,12 @@ func InvokeGenerationFromSheet(wg *sync.WaitGroup) error {
 		utils.ProcessModelToComponentsMap(componentCSVHelper.Components)
 		utils.VerifyandUpdateSpreadsheet(spreadsheeetCred, &wgForSpreadsheetUpdate, srv, spreadsheeetChan, spreadsheeetID)
 	}()
-
 	// Iterate models from the spreadsheet
 	for _, model := range modelCSVHelper.Models {
+		if modelName != "" && modelName != model.Model {
```

Comment:
I think it would be more efficient if we can parse only the models with the given name in the first place

Reply:
Gents, pick a path forward at-will, and proceed. Either way, SGTM.